zebra-chain/src/sapling/shielded_data.rs

```diff
@@ -199,7 +199,6 @@ where
         // TODO: use TransferData::shared_anchor to improve performance for V5 transactions
         self.spends_per_anchor()
             .map(|spend| spend.per_spend_anchor)
-            .sorted()
```

Comment:
We can't derive `Sort` with jthe new type as `jubjub::Fq` does not implement this. By this i had to remove the `sorted()` here. All tests keep passing so i am not sure of the impact on this.

Reply:
This is a good question!

Skipping `sorted()` could return some duplicate anchors. This makes us validate some anchors multiple times, which slows down validation for large transactions.

I didn't check all the code that calls this method yet.

To keep `sorted()`, we could implement `Ord` and `PartialOrd` manually for `Root`, using `jubjub::Fq`'s raw bytes method?

I couldn't find a good example, but the code might look something like this:
https://github.com/ZcashFoundation/zebra/blob/4aeabd0b5272607b730e0bd283c88a49294301cc/zebra-state/src/service/non_finalized_state/chain.rs#L958-L977

Reply:
So i added the `unique_by` option at https://github.com/ZcashFoundation/zebra/pull/3544/commits/4a8509244fbdaadcbb2d7820b9ab17a1c95f0131 however i need to confirm that is what we need here.

Reply:
This looks good, I made one suggestion to simplify the code.

And yes, we want to compare the bytes here.